adagrams/game.py

```diff
@@ -1,11 +1,168 @@
+            letter_frequency[letter] += 1
+        else:
+            letter_frequency[letter] = 1
+        if letter_frequency[letter] <= LETTER_POOL[letter]:
+            letters.append(letter)
+        if len(letters) == 10:
+            break
+    print(len(letters))
+    return letters
+    
+    
+##-----------------------------------------##
+##-----------------WAVE 2------------------##
+##-----------------------------------------##
+"""
+2. You need a way to check if an input word (a word a player submits) only uses characters that 
+    are contained within a collection (or hand) of drawn letters
+-Has two parameters:
+    -word, the first parameter, describes some input word, and is a string
+    -letter_bank, the second parameter, describes an array of drawn letters in a hand. You can expect this to be an array of ten strings,
+    with each string representing a letter
+-Returns either True or False
+-Returns True if every letter in the input word is available (in the right quantities) in the letter_bank
+-Returns False if not; if there is a letter in input that is not present in the letter_bank or has too much of compared to the letter_bank
+reference: https://www.programiz.com/python-programming/methods/list/count
+"""
 
 def uses_available_letters(word, letter_bank):
-    pass
+    # ensure consistent case
+    word = word.upper()
+    # check input word is anagram within drawn letters
+    #iterating over each letter in the word
+    for letter in word:
+        # check if that letter is in the letter bank
+        if letter in letter_bank:
+            word_count = word.count(letter)
+            # if word_count is in the right quantities in the letter bank, continue checking next letter
+            if word_count == letter_bank.count(letter):
+                continue
+            else:
+                return False
+        else:
+            return False
+    return True
+
 
+##-----------------------------------------##
+##-----------------WAVE 3------------------##
+##-----------------------------------------##
+"""
+3. Now you need a function returns the score of a given word as defined by the Adagrams game.
+-Has one parameter: word, which is a string of characters
+-Returns an integer representing the number of points
+-Each letter within word has a point value. The number of points of each letter is summed up to represent the total score of word
+-Each letter's point value is described in the table below
+-If the length of the word is 7, 8, 9, or 10, then the word gets an additional 8 points
+"""
 def score_word(word):
-    pass
+    score_chart = {"A":1, "E":1, "I":1, "O":1, "U":1, "L":1, "N":1, "R":1, "S":1, "T":1, "D":2,
+    "G":2, "B":3, "C":3, "M":3, "P":3, "F":4, "H":4, "V":4, "W":4, "Y":4, "K":5, "J":8, "X":8,
+    "Q":10, "Z":10
+    }
+    score = 0
+    if len(word) >=7:
+            score += 8
+    for letter in word:
+        score += score_chart[letter.upper()]
+            
+    return score
+
```

Comment:
This looks great! One small extension would be to include a check on line 102 that only adds the 8 points if the word is also less than 11 letters long.

Also, just in terms of python best practices, in an instance like this, it's usually more customary to put the for loop first!